docs/release-procedure.md

```diff
@@ -23,29 +24,49 @@ as reviewing them again would not add a lot of value but would add lots of exces
 Hence non-cherry-pick commits should happen in a separate branch in your own fork, and
 be delivered to the release branch by using a PR.
 
-Note that Modin uses fully signed commits at least for releases, so you have to have GPG keys set up. See [onboarding instructions](https://github.com/modin-project/modin/blob/master/onboarding/onboarding.md) on where to get started.
+Note that Modin uses fully signed commits, so you have to have GPG keys set up. See [onboarding instructions](https://github.com/modin-project/modin/blob/master/onboarding/onboarding.md) on where to get started.
 
 To update Modin version, follow the instructions below.
 
-### Update the link to release in readme
+### Preparing the repo for a Major or Minor Version
 
-**Note**: from now on you work in a branch in your fork (in `origin`).
+**Note**: this should be done in your fork of Modin.
 
-Change the `README.md` in the repo root so release badge points to new (would be created) release.
+First, update your fork of Modin's master with the main repo's master. From your master, create a new
+branch called `release-X.Y.0` off of master. Create an empty commit in your new branch with the message
+`Release version X.Y.0`. Make sure to sign this commit with both your GPG key
+and with the conventional `git commit -s` (so `git commit -s -S`). Open a PR against modin-project/modin with just this commit.
 
-### Commit
+### Preparing the repo for a Patch Version
 
-        git commit -a -S -m "Release version X.Y.Z"
+**Note**: this should be done in the original Modin repository (in `upstream`) .
 
-Pay attention to that `-S` switch if you haven't enabled signing all the commits!
-Modin requires release commits to be signed (do not mix this with "signing off" commits, this
-is a whole different story - signed commits are cryptographically signed).
+First, you must create a new branch in the upstream (main modin-project/modin) repo for the new release.
+This branch must be named `release-X.Y.Z`, and should be made off of the tag for the last release. To
+do this, look up the commit that corresponds to the last release, copy its SHA, and then do 
+`git checkout <PREVIOUS_RELEASE_COMMIT_SHA>`. Next, do `git checkout -b release-X.Y.Z` to create
+the branch for the new release, and push it upstream.
 
-Push the commit, make a PR _against `release-X.Y.Z`_ branch and, when it's merged, pull that branch from `upstream` before proceeding.
+**Note**: now you must switch to your fork of Modin.
+
+From your fork of Modin, fetch the upstream repo, and checkout the release branch you made above.
+From this release branch, create a new branch, and cherry-pick the commits that will go into this release.
```

Comment:
why do we want to cherry-pick in fork instead of upstream repo? this IMHO makes reviewing release PRs somewhat harder without actually gaining a lot of value